### DISCUSSION TITLE
Discourse forum for Modrinth

### DISCUSSION THREAD

**Discussion by brawaru** [modrinth/.github#147] [Feature requests]
(This post intends to supersede #143; everyone is invited to participate in the discussion through replies!)

## The current state of things

Modrinth has a amazing and vibrant community on Discord. I feel it's important for Modrinth to continue to maintain a space for this community to come along. However, one of the primary goals of Modrinth's Discord server is to provide support with Modrinth to users and creators, not to be a community hub. This leads to several challenges.

As a moderator, I've noticed a recurring frustration where people ask the same questions repeatedly, only to receive the same responses that were previously provided to others or a generic response using a tag. This often happens because only active community members are aware of prior discussions.

Another issue is the lack of clear direction. Newcomers to Modrinth's Discord often jump into the first channel they see to ask their questions. Then they are redirected to various channels like #lounge, #mod-support, #website-support, #app-support, #gayming, #technology, or asked to contact the Modmail bot. While this behaviour is partially understandable due to the way Discord encourages immediate interaction, the first issue exacerbates the situation. Even if someone joins and reads welcome messages, finding useful information through Discord's search feature can be challenging.

A third issue revolves around organization and overall chaos, which became apparent during the app's alpha testing phase. Initially, two dedicated channels existed – one for announcements and another for feedback. However, with the introduction of an additional testing server for the app's alpha testing, a new dynamic emerged. Unlike previous tests, feedback and bug reports for the app needed to be submitted on GitHub [^1]. Yet, users accustomed to the old system continued flooding feedback channels with unrelated feedback and bug reports for the app, which was incredibly frustrating as a moderator, and even led to creation of a special form closer to the end of alpha testing, which directed users to the right channels.

The current situation is both exhausting and frustrating, and it's clear that a better solution is needed.

## A forum at the end of the day?

I've noticed that many platforms have adopted forums as a solution. Unlike old clunky phpBB Invision XenForo forums, modern forums, such as Discourse, NodeBB, and Flarum, offer a more streamlined approach. They feature a continuous stream of posts organized by category and tagged for easy searchability. A forum could serve as a potentially superior alternative to Discord.

Forums offer the ability to make announcements, polls, easily re-categorise posts, and allow indexing by search engines. Tags, for example, can be instrumental in tracking the status and relevance of feedback issues. Categories and subcategories can cover various topics, from support and information hubs to showcases and general discussions. Additional tools like PSAs (banners at the top of the site) and pinned posts can provide users with essential information before they start posting.

Considering that Modrinth now uses its own authentication system, integration with these forums should be feasible. This integration would allow new users to seamlessly sign in on the forum using their existing Modrinth accounts. The familiar sign-in process could encourage increased engagement within the forum community. Using the Modrinth account on the forum could also make providing support easier, since the account is immediately visible without needing to ask questions.

Moreover, these forums often provide robust moderation tools with granular permissions. Community moderation features allow users to collectively address negative behaviour, potentially minimising harm before a moderator intervenes.

While adopting a forum presents its challenges, such as deployment and administration, it offers a more comprehensive solution compared to a Discord server. Outsourcing the forum's management is an option, although it may come at a higher cost than manual management. Of them all, Discourse appears to be the most mature and best option — it can be both [self-hosted](https://github.com/discourse/discourse) and [outsourced](https://discourse.org/pricing).

## Join the discussion

I really want to hear your thoughts. As users, would you welcome the introduction of a forum? Would you actively use it? Share your current frustrations with Modrinth's community organisation, and whether the forum would address them. Your input is incredibly valuable, as different perspectives can lead to better solutions. If you're definitely in favour of having forums, please leave an upvote on this post.

[^1]: This was not a problem with using GitHub for this purpose. In fact, GitHub was a much better solution than having all feedback in a messy forum channel. It's just that this change was introduced and timed rather badly, which created a lot of confusion and frustration.

**Comment by thewilloftheshadow** [comment]
https://meta.discourse.org/t/set-up-discord-notifications-with-the-discourse-chat-integration-plugin/66600 :grin:

**Reply by brawaru** [reply]
I saw that, yeah, that is definitely can be useful!

**Comment by 4vyy** [comment]
I think it would definitely be a good idea to be less dependent on Discord in terms of community and info as Modrinth grows. However, I'm unsure what exactly this would mean for the Modrinth Discord.

The two main cons to a forum over a chatroom like Discord that I can think of is that it's independent and can have less interaction/overlap with other communities on Discord and also that there's no lively, flowing chat for people to interact and community to foster.

What seems the most beneficial solution is that the forum would be for discussion and support of and around Modrinth and the Discord server be _much_ more simple with just general discussion (though probably keep #modrinth-discussion to still have a general chat about Modrinth as a whole) and probably still all the announcement/feed channels. Overall, it seems like a good idea and would make guiding users less chaotic.

**Comment by Andre601** [comment]
Personally, I dislike this idea...

The main issue I see is, that this won't fix the current issues but rather just shift them to a separate place to deal with. This would in the end also increase the moderation work as there would be yet another location mods need to check (Given that mods on Discord would also moderate the forums).

You may argue that the forum would solve issues such as repeated questions, but I bet to differ here. I doubt that this will truly be the case and that duplicate questions with the same answer will still be a daily thing. People are lazy and look for a path with the least struggle, which in this case would be not searching for your question and just ask anyways.

Also, this here is a personal thing, so it's subjective in nature, but I dislike the style some forum softwares have... Primarely Discourse.
It may look simple and probs tries to achieve a material-like theme here? But for me does it look like someone without UI experience tried to make a forum software from scratch... It looks bland and unappealing. I never was a fan of it in the first place as it just felt cheap and not like quality being given here...

And lets be honest: There are some people who like Modrinth for its nice looking Website design, so going to a forum that looks nothing alike and perhaps even bad in comparison would drive people away. Sure, you can modify the CSS and source itself, but that is extra work the Modrinth team would need to put into something that in my opinion doesn't even give you anything better that Discord and GitHub aren't providing already.

Also, one thing I personally dislike too about Discourse is, how you do the exact same tutorial on the exact same features over and over again whenever you register a new account.
It drains you of any motivation if you have to go through that same tutorial for the 30th time for stuff you already know and could skip (And sadly, some forums force you to make it to actively chat in some of the forums).

If anything then Modrinth should finally create a FAQ page where the most common questions are answered... This *could* help people in the long run here eventually.

**Reply by brawaru** [reply]
The issue not repetitive questions per se, but that Discord is currently used as an information hub, which it is terrible at, and you will only go as far with the proper documentation, it will never cover everything, and people will always ask questions. If people have access to information, which in this case — answers to others, it will certainly reduce the amount of similar questions. Modrinth has moderation delays? There's a pinned post about that. Modrinth is currently down? Pinned post and a huge banner at the top of the page. User asked something already answered? Just forward them another post's message. They asked question in the wrong place? Just recategorise the post.

As for moderation, with forums you take a reactive moderation approach, you react to reports rather than proactively checking everything like you do on Discord. You can still actively moderate, but it's not necessary. For me personally moderating forum wouldn't much of an issue, in fact, it might be significantly better with proper moderation toolchain usually provided by such software. Moreover, since things like Discourse are flat forums, you immediately see all posts, without need to go into every channel and checking individual messages. The way people post on forums also actively differs, so it's easily to grasp context. I cannot say for certain, I never tried, but so far it looks significantly easier and better than moderating Discord.

Theming: I don't think it would be too much of a concern and that many would actually care. Discourse allows customisability, see [Figma Community](https://forum.figma.com/), [Quilt](https://forum.quiltmc.org/), and theming category on [Discourse meta](https://meta.discourse.org/) and [their customers page](https://www.discourse.org/). It also has good DX without things like full page reloads, pretty similar to Modrinth. If you are on it, why Modrinth documentation and blog websites having three different styles never bothered you?

Tutorial: first of all, how many accounts are you planning to register on Modrinth? I thought 1 was enough for majority of the users? And then, I'm not very sure what tutorial are you talking about, unlike full screen onboarding in Modrinth app, Discourse only has some tooltips here and there (which are great way to introduce new users to how things are being done) and a bot message. That's pretty much it. Don't like it? You can turn it off in settings. There's of course leveling system, but it's not too much of a bother and probably can be tweaked.

That being said, I don't agree with dismissal of this idea purely because Discourse has different UX and design. And as I said above, Modrinth documentation will only get you so far. It won't cover unrelated to Modrinth stuff, it won't be able to cover all issues, it won't be always up-to-date, because that's simply impossible. Discord doesn't cut it. So forum has its place to be. But that's what I think, everyone gets their perspective and that's okay. At the end of the day, no one wanted to get rid of Discord, not yet. Discord is great for community organisation, but it's absolutely terrible at getting information through and providing support.

**Reply by thewilloftheshadow** [reply]
By tutorial they mean the tutorial on Discourse that you get when you register a new account.

**Reply by Andre601** [reply]
Bot dm about the basics was what I meant.

**Comment by un-pogaz** [comment]
Like Andre601, I don't think it would really solve the problem but rather move it... but I think creating a Forum would be a better solution than Discord, because it would move it to a place _designed to handle this kind of problem_.

In fact, my message is going to be to give more weight to your proposal with arguments that appeared to me after reading your message.

As you said, when people come to Discord, they immediately post their problems with a minimum of research... and that's normal: the purpose of Discord _is_ the direct interaction and rapid response. Yes, there is a Pinned Message system, but it's not promoted at all, it's anecdotal and nobody looks at it.
A Forum, by its very nature, is slower and quieter. People are much more inclined to look at previous topics, to read previous conversations, to do research if necessary and the advanced search is much more accessible (even if it's still a keyword system like Discord, Github Reddit... there's a GUI). And, if a user wants to create a new post, they will tend to create a larger and more complete post on their problem, which will make it easier to solve.
So, Yes, there's now a forum system in Discord, but it's a copy of a real forum system, I've always seen that as Discord, just in a different CSS.

So, it won't solve the problem of duplicates of guys who post super short messages without searching, but it will help.
What's more, we'll be able to benefit from the extra strike force of super-users who have no life and spend all day on the forum and who, in reply n°1, will give a link to a previous conversation on the topics "closing" the conversation before a moderator arrives.

Another point that will also be incredibly useful is more efficient moderation. So I'm not talking about greater granularity in the tools, roles and power given (you've presented that well, and Discord is pretty good at that too), but I'm talking about the fact that we'll probably have more people.
Why? Same that previous, because a forum is slower. Discord is all about live, frenetic conversation, so it's very user-friendly, but when it comes to moderation, you have to be so much more reactive, you have to be constantly on the alert for anything that goes wrong, it's exhausting and off-putting. For a Forum, it's possible that moderators will be much easier to find, because they know they'll be able to take more time and work at their own pace more efficiently, rather than constantly being in the rush of Discord's chaotic frenzy.
I think that with the same number of moderators, a Forum is better moderated than Discord (personaly, I'd be much more inclined to moderate a Forum than a Discord server).

The third point that comes to mind, but which will be inevitable in some time, is control and sovereignty.
Discord, however friendly it may be, is and remains a service provided by a third party, in proprietary software managed by a capitalist company, whose policies and objectives are uncertain and can be unhealthy.
For a site that wants to be open-source and respectful of its users... yea- No. That not match very good.
Well, I'm not going to do what some FOSS fanatics do and call you a hypocrite, because I simply understand that setting up a forum is an important infrastructure, that it's not your priority and, also, you have the right to be lazy (proprietary software counts on this laziness, but you have the right to do). Just as long as you have the time and energy, it will be appreciated, thank you.

Discord will never disappear, live chat has undeniable strengths, but with a Forum, it will certainly diminish considerably in size to be concerning only a hard core and very involved people, while the bulk of users will be parked in the Forum with its own ecosystem to regulate it.

 

Finally, a more... technically daring point:
It's already quite clear that at some point Modrinth will integrate a comment system into the project pages. I think that if we get things right, we could kill two birds with one stone by simultaneously creating the Forum and project comments.
Here's the idea: When a project page is created, a Forum post is automatically created and linked to it. And the reply section of this Forum post will be imported and used in the project page comment section. And it will be possible to browse the Projects, through the Forum interface (for those who wish to do so), it will just be necessary for the Master-Post to auto-load the body of the project page.
One of the advantages of using the forum to create the comment section of a project is that the management and moderation of this part of Modrinth under the jurisdiction of the Forum moderation team without having to specifically add the role of 'comment moderator' and all that goes with it. All moderation can be done transparently via the Forum interface tools.
So, it would be a bit complicated because you'd have to give moderation powers to project creators on their own Project page/Auto-Post so that they can manage their comment section... but we're God on our forum, probably we can implemented that.

**Comment by brawaru** [comment]
This is screenshot of Google Trends from 19.08.23 just confirming the importance of having an open and accessible source for information like the recent authorisation issues. The users that googled this would never receive an answer because Discord servers are essentially walled gardens.

![Screenshot of related queries on Google Web Trends sorted by Rising, the list consists of: 1. 500 internal server error (breakout); 2. figure mod (breakout); 3. tweakaroo (breakout); 4. ely by (breakout) ; 5. better than adventure (+250%)](https://github.com/modrinth/.github/assets/10401817/c040ad30-d596-45b5-8e43-951c5178166c)

**Comment by osfanbuff63** [comment]
I would support this, as Discord being a hub for everything isn't a great idea, and as mentioned above a search for '500 internal server error' might show a Modrinth forum result, but not a Discord result.
